app/services/billable_metrics/aggregations/recurring_count_service.rb

```diff
@@ -63,6 +76,38 @@ def aggregation_query
         "SELECT (#{queries.map { |q| "COALESCE((#{q}), 0)" }.join(' + ')}) AS aggregation_result"
       end
 
+      def group_aggregation_query
+        group_queries = groups.each_with_object([]) do |group, result|
+          result << [
+            # NOTE: Billed on the full period
+            persisted.select(
+              "(SUM(#{persisted_pro_rata}::numeric)) as group_sum, #{sanitized_name(group)} as group_name",
+            ).group(sanitized_name(group)).to_sql,
+
+            # NOTE: Added during the period
+            added.select(
+              "(SUM(('#{to_date}'::date - DATE(persisted_events.added_at) + 1)::numeric  / #{period_duration})::numeric) as group_sum, \
```

Comment:
Layout/LineLength: Line is too long. [136/120] (https://rubystyle.guide#max-line-length)